File: custom_app/public/js/item.js
frappe.ui.form.on('Item', {
    refresh: function(frm) {
        if (frm.doc.__islocal) {            
            return;
        }

        frm.add_custom_button(__('Send Email'), function() {
            sendEmail(frm);
        }, __('Actions'));
    }
});

function sendEmail(frm) {
    frappe.call({
        method: 'custom_app.public.py.item.send_email',
        args: {
            item_code: frm.doc.item_code
        },
        callback: function(response) {
            if (response.message) {
                frappe.msgprint(__('Email sent successfully.'));
            } else {
                frappe.msgprint(__('Failed to send email.'));
            }
        }
    });
}

File: custom_app/public/py/item.py
 
 
import frappe

@frappe.whitelist()
def send_email(item_code):
    sql_query = """
        SELECT
            si.name,
            si.posting_date
        FROM
            `tabSales Invoice` si
        INNER JOIN
            `tabSales Invoice Item` item ON si.name = item.parent
        WHERE
            item.item_code = %(item_code)s
        ORDER BY
            si.posting_date DESC
        LIMIT 5
    """

    sales_invoices = frappe.db.sql(sql_query, {'item_code': item_code}, as_dict=True)

    if not sales_invoices:
        frappe.msgprint('No sales invoices found for this item.')
        return

    attachments = []
    for invoice in sales_invoices:
        attachment_content = get_pdf_data('Sales Invoice', invoice['name'])
        attachments.append({
            'fname': f'{invoice["name"]}.pdf',
            'fcontent': attachment_content
        })

    message = frappe.render_template('custom_app/templates/emails/sales_invoice_template.html', {
        'item_name': item_code,
        'sales_invoices': sales_invoices
    })

    frappe.sendmail(
        recipients=['kumvipul1047@gmail.com'],
        subject=f'Latest Sales Invoices for Item: {item_code}',
        message=message,
        attachments=attachments,                
    )
    return True

def get_pdf_data(doctype, name):     
    html = frappe.get_print(doctype, name)
    return frappe.utils.pdf.get_pdf(html)

 
 